jupyterlab/commands.py

```diff
@@ -253,10 +308,70 @@ def uninstall_extension(name, app_dir=None):
 
 
 def list_extensions(app_dir=None):
-    """List installed extensions.
+    """List the extensions.
     """
     app_dir = get_app_dir(app_dir)
-    return sorted(_get_extensions(app_dir).keys())
+    extensions = _get_extensions(app_dir)
+    disabled = _get_disabled(app_dir)
+    linked = _get_linked_packages(app_dir)
+    app = []
+    sys = []
+    linked = []
+
+    # We want to organize by dir.
+    sys_path = pjoin(get_app_dir(), 'extensions')
+    for (key, value) in extensions.items():
+        if key in linked:
+            linked.append(key)
+        if value['path'] == sys_path and sys_path != app_dir:
+            sys.append(key)
+            continue
+        app.append(key)
+
+    print('Known labextensions:')
+    if app:
+        print('   app dir: %s' % app_dir)
+        for item in sorted(app):
+            extra = ''
+            if item in linked:
+                extra += '*'
+            if disabled.get(item, False):
+                extra += ' %s' % RED_DISABLED
+            else:
+                extra += ' %s' % GREEN_ENABLED
+            print('        %s%s' % (item, extra))
+
+    if sys:
+        print('   sys dir: %s' % sys_path)
+        for item in sorted(sys):
+            extra = ''
+            if item in linked:
+                extra += '*'
+            if disabled.get(item, False):
+                extra += ' %s' % RED_DISABLED
+            else:
+                extra += ' %s' % GREEN_ENABLED
+            print('        %s%s' % (item, extra))
+
+    if linked:
+        print('* Denotes linked packages')
+
+    # Handle uninstalled and disabled core packages
+    uninstalled_core = _get_uinstalled_core_extensions(app_dir)
+    if uninstalled_core:
+        print('\nUninstalled core extensiosn:')
+        [print(item) for item in sorted(uninstalled_core)]
+
+    core_extensions = _get_core_extensions()
+
+    disabled_core = []
+    for key in core_extensions:
+        if disabled.get(key, False):
+            disabled_core.append(key)
+
+    if disabled_core:
+        print('\nDisabled core extensions:')
+        [print(item) for item in sorted(disabled_core)]
```

Comment:
I would indent these extension by 4 spaces and follow it all by a newline. This part of the output blended into following lines in the terminal making it tough to detect.